src/VisualStudio/Core/Def/Implementation/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -46,77 +47,97 @@ private enum GlobalNotificationState
+                    {
+                        client.Dispose();
+                    }
                 }
 
-                void Shutdown(Exception ex)
-                {
-                    // make sure we shutdown client if initializing client has failed.
-                    client?.Shutdown();
+                client.Started();
+                return client;
+            }
+        }
 
-                    // translate to our own cancellation if it is raised.
-                    cancellationToken.ThrowIfCancellationRequested();
+        public static async Task<Stream> RequestServiceAsync(
+            Workspace workspace,
+            HubClient client,
+            string serviceName,
+            HostGroup hostGroup,
+            CancellationToken cancellationToken)
+        {
+            Exception unexpectedException;
 
-                    // otherwise, report watson
-                    ex.ReportServiceHubNFW("ServiceHub creation failed");
-                }
+            var descriptor = new ServiceDescriptor(serviceName) { HostGroup = hostGroup };
+            try
+            {
+                return await client.RequestServiceAsync(descriptor, cancellationToken).ConfigureAwait(false);
+            }
+            catch (Exception e) when (e is ConnectionLostException || e is RemoteInvocationException re && re.InnerException is InvalidOperationException)
+            {
+                // ServiceHub may throw these exceptions if it is called after VS started to shut down.
+                // Cancel the operation and do not report an error in these cases.
+                // 
+                // TODO: Once https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1040692 is implemented,
+                // i.e. the ServiceHub does not throw these exceptions when cancellation token is signaled,
+                // we can assume that these exceptions indicate a failure and should be reported to the user.
+                // That's because we expect the cancellation to be requested on the token before HubClient gets disposed.
+                cancellationToken.ThrowIfCancellationRequested();
+
+                unexpectedException = e;
             }
+            catch (Exception e) when (!(e is OperationCanceledException))
+            {
+                unexpectedException = e;
+            }
+
+            throw SoftCrash(workspace, client, unexpectedException, "RequestServiceAsync Failed", cancellationToken);
+        }
+
+        private static SoftCrashException SoftCrash(Workspace workspace, HubClient client, Exception unexpectedException, string message, CancellationToken cancellationToken)
+        {
+            client.Logger.TraceEvent(TraceEventType.Error, 1, unexpectedException.ToString());
+            unexpectedException.ReportServiceHubNFW(message);
+            RemoteHostCrashInfoBar.ShowInfoBar(workspace, unexpectedException);
+            return new SoftCrashException(message, unexpectedException, cancellationToken);
```

Comment:
is SoftCrashException just a subclass of OperationCanceledException?  or do higher up levels have special knowledge to ignore it?